src/tools/data_handlers/analytics_handler.py

```diff
@@ -380,10 +399,29 @@ async def pull_data(
                     )
 
             if "status" in result and result["status"] in ["success", "saved"]:
-                raw_data, data_points_count, message_detail = (
-                    self._process_response_data(result)
-                )
-
+                (
+                    raw_data,
+                    data_points_count,
+                    message_detail,
+                ) = await self._process_response_data(result)
+                logger.debug(f"RAW_DATA: {raw_data}")
+
+                # Enrich raw_data with names
+                if subregion:
+                    subregion_aois_id_to_name = {
+                        format_id(item["src_id"]): item["name"].split(",")[0]
+                        for item in subregion_aois
+                    }
+                    # For tree cover loss, the API returns "id" as the key for the AOI IDs, for other datasets it returns "aoi_id"
```

Comment:
Seems like something to cahnge in the api design. We should report this to Justin and team.

Reply:
yes, I have reported this to him over slack